transformer_engine/common/include/transformer_engine/permutation.h

```diff
@@ -18,4 +18,8 @@ void nvte_unpermute(const NVTETensor input, NVTETensor output, NVTETensor row_id
                     const NVTETensor prob, const int num_rows, const int topK, const int num_cols,
                     cudaStream_t stream = nullptr);
 
+void nvte_device_radix_sort_pairs(NVTETensor temp_storage, size_t &temp_storage_bytes,
+                                  NVTETensor indices, NVTETensor sorted_indices, NVTETensor row_id,
+                                  NVTETensor sorted_row_id, int64_t max_expanded_token_num);
```

Comment:
Minor issues:
- C does not support pass-by-reference
- `size_t` is more logical for `max_expanded_token_num` than `int64_t`
- It would be nice to tweak variable names to keep this function generic and match [`cub::DeviceRadixSort::SortPairs`](https://nvidia.github.io/cccl/cub/api/structcub_1_1DeviceRadixSort.html#_CPPv4I000EN3cub15DeviceRadixSort9SortPairsE11cudaError_tPvR6size_tPK4KeyTP4KeyTPK6ValueTP6ValueT9NumItemsTii12cudaStream_t):
```suggestion
void nvte_device_radix_sort_pairs(NVTETensor temp_storage, size_t *temp_storage_bytes,
                                  NVTETensor keys_in, NVTETensor keys_out, NVTETensor values_in,
                                  NVTETensor values_out, size_t num_items);
```